tests/integration/ha_tests/test_smoke.py

```diff
@@ -0,0 +1,173 @@
+    """Test charm deploy in dirty environment with garbage storage"""
+    async with ops_test.fast_forward():
+        garbadge_storage = None
+        for attempt in Retrying(stop=stop_after_delay(30 * 3), wait=wait_fixed(3)):
+            with attempt:
+                garbadge_storage = await get_any_deatached_storage(ops_test)
+                assert garbadge_storage is not None
+
+        assert garbadge_storage is not None
+
+        await ops_test.model.applications[APPLICATION_NAME].add_unit(1, attach_storage=[tag.storage(garbadge_storage)])
+
+        # Reducing the update status frequency to speed up the triggering of deferred events.
+        await ops_test.model.set_config({"update-status-hook-interval": "10s"})
+
+        await ops_test.model.wait_for_idle(apps=[APPLICATION_NAME], status="active", timeout=1500)
+        assert ops_test.model.applications[APPLICATION_NAME].units[0].workload_status == "active"
+
+        primary_name = await get_primary(ops_test, ops_test.model.applications[APPLICATION_NAME].units[0].name)
+        assert await is_postgresql_ready(ops_test, primary_name)
+
+        storage_id_str = storage_id(ops_test, primary_name)
+
+        # Check if storage exists after application deployed
+        assert await is_storage_exists(ops_test, storage_id_str)
+
+        # Check that test database is not exists for duplicate application 
+        assert not await check_db(ops_test, APPLICATION_NAME, TEST_DATABASE_RELATION_NAME)
+
+        await ops_test.model.destroy_unit(primary_name, destroy_storage=False, max_wait=1500)
+
+
+@pytest.mark.group(1)
+@pytest.mark.abort_on_fail
+async def test_app_recoures_conflicts(ops_test: OpsTest, charm: str):
+    """Test application deploy in dirty environment with garbage storage from another application """
+    async with ops_test.fast_forward():
+        garbadge_storage = None
+        for attempt in Retrying(stop=stop_after_delay(30 * 3), wait=wait_fixed(3)):
+            with attempt:
+                garbadge_storage = await get_any_deatached_storage(ops_test)
+                assert garbadge_storage is not None
+
+        assert garbadge_storage is not None
+
+        # Deploy duplicaate charm
+        await ops_test.model.deploy(
+            charm,
+            application_name=DUP_APPLICATION_NAME,
+            num_units=1,
+            series=CHARM_SERIES,
+            config={"profile": "testing"},
+            attach_storage=[tag.storage(garbadge_storage)]
+        )
+
+        try:
+            await ops_test.model.wait_for_idle(apps=[DUP_APPLICATION_NAME], timeout=500, status="blocked")
+        except (TimeoutError) as e:
+            logger.info(f"Application is not in blocked state. Checking logs...")
+
+        # Since application have postgresql db in storage from external application it should not be able to connect due to new password
+        assert not await check_password_auth(ops_test, ops_test.model.applications[DUP_APPLICATION_NAME].units[0].name)
```

Comment:
This charm stalls forever in Init:CrashLoopBackOff loop with "waiting, installing agent". As I know, this [commit](https://github.com/canonical/postgresql-operator/commit/a18b1d3f951c02b6a41b1ab041b3856421476e74) will fix this issue, and deployment will be in `blocked` state instead of falling in to forever CrashLoopBackOff due to database from another application.